test/odbc/psqlodbc/float.cpp

```diff
@@ -0,0 +1,458 @@
+  const vector<string> NAME_EXPECTED = {"int4", "float8"};
+
+  createTable(ServerType::PSQL, TABLE_NAME, TABLE_COLUMNS);
+  testCommonColumnAttributes(ServerType::PSQL, TABLE_NAME, TABLE_COLUMNS.size(), COL1_NAME, LENGTH_EXPECTED, 
+    PRECISION_EXPECTED, SCALE_EXPECTED, NAME_EXPECTED);
+  dropObject(ServerType::PSQL, "TABLE", TABLE_NAME);
+}
+
+TEST_F(PSQL_DataTypes_Float, Table_Create_Fail) {
+  const vector<vector<pair<string, string>>> invalid_columns {
+    {{"invalid1", DATATYPE_NAME + "(0)"}} // Cannot specify a column width on data type float.
+  };
+
+  // Assert that table creation will always fail with invalid column definitions
+  testTableCreationFailure(ServerType::PSQL, TABLE_NAME, invalid_columns);
+  dropObject(ServerType::PSQL, "TABLE", TABLE_NAME);
+}
+
+TEST_F(PSQL_DataTypes_Float, Insertion_Success) {
+  double data{};
+  const int bufferLen = 0;
+
+  const vector<string> VALID_INSERTED_VALUES = {
+    FLOAT_15,
+    FLOAT_383,
+    "0.4347509234",
+    "NULL"
+  };
+
+  const vector<double> expected = getExpectedFloatResults(VALID_INSERTED_VALUES);
+
+  const vector<long> expectedLen(expected.size(), FLOAT_BYTES_EXPECTED);
+
+  createTable(ServerType::PSQL, TABLE_NAME, TABLE_COLUMNS);
+  testInsertionSuccess(ServerType::PSQL, TABLE_NAME, COL1_NAME, SQL_C_DOUBLE, data, bufferLen, VALID_INSERTED_VALUES, 
+    expected, expectedLen);
+  dropObject(ServerType::PSQL, "TABLE", TABLE_NAME);
+}
+
+TEST_F(PSQL_DataTypes_Float, Insertion_Failure) {
+  // Range -1e308 to -1e-307  0 and 0 1e307 to 1e308 
+  const vector<string> INVALID_INSERTED_VALUES = {
+    "1e309",
+    "-1e309",
+    "1e-325",
+    "-1e-324"
+  };
+
+  createTable(ServerType::PSQL, TABLE_NAME, TABLE_COLUMNS);
+  testInsertionFailure(ServerType::PSQL, TABLE_NAME, COL1_NAME, INVALID_INSERTED_VALUES, true);
+  dropObject(ServerType::PSQL, "TABLE", TABLE_NAME);
+  }
+
+
+TEST_F(PSQL_DataTypes_Float, Update_Success) {
+  double data;
+  const int bufferLen = 0;
+
+  const vector<string> DATA_INSERTED = {"0.1"};
+  const vector<double> data_expected = getExpectedFloatResults(DATA_INSERTED);
+  const vector<long> expectedInsertLen(DATA_INSERTED.size(), FLOAT_BYTES_EXPECTED);
+
+  const vector<string> DATA_UPDATED_VALUES = {
+    FLOAT_383,
+    "0",
+    FLOAT_15
+  };
+  const vector<double> DATA_UPDATED_EXPECTED = getExpectedFloatResults(DATA_UPDATED_VALUES);
+
+  const vector<long> expectedLen(DATA_UPDATED_EXPECTED.size(), FLOAT_BYTES_EXPECTED);
+  
+  createTable(ServerType::PSQL, TABLE_NAME, TABLE_COLUMNS);
+  testInsertionSuccess(ServerType::PSQL, TABLE_NAME, COL1_NAME, SQL_C_DOUBLE, data, bufferLen, DATA_INSERTED, data_expected, expectedInsertLen);
+  testUpdateSuccess(ServerType::PSQL, TABLE_NAME, COL1_NAME, COL2_NAME, SQL_C_DOUBLE, data, bufferLen, DATA_UPDATED_VALUES, 
+    DATA_UPDATED_EXPECTED, expectedLen);
+  dropObject(ServerType::PSQL, "TABLE", TABLE_NAME);
+}
+
+TEST_F(PSQL_DataTypes_Float, Update_Fail) {
+  double data;
+  const int bufferLen = 0;
```

Comment:
nit: constants should be all capitals. This goes with all `bufferLen` in the file.

Reply:
Done. Hopefully I caught them all.